planetiler-core/src/main/java/com/onthegomap/planetiler/reader/SimpleReader.java

```diff
@@ -49,15 +54,18 @@ protected SimpleReader(Profile profile, Stats stats, String sourceName) {
   public final void process(FeatureGroup writer, PlanetilerConfig config) {
     var timer = stats.startStage(sourceName);
     long featureCount = getCount();
+    int readThreads = config.featureReadThreads();
     int writeThreads = config.featureWriteThreads();
     int processThreads = config.featureProcessThreads();
     Envelope latLonBounds = config.bounds().latLon();
     AtomicLong featuresRead = new AtomicLong(0);
     AtomicLong featuresWritten = new AtomicLong(0);
 
     var pipeline = WorkerPipeline.start(sourceName, stats)
-      .fromGenerator("read", read())
-      .addBuffer("read_queue", 1000)
+      .readFrom("source_paths", sourcePaths)
+      .addBuffer("read_queue", 1000, 1)
```

Comment:
This allocates a new thread to iterate through the list, we could skip that by either moving the logic to iterate through the paths into the read thread, or change it to use `readFromTiny` which just copies every element from sourcePaths into a queue and makes that the input to the next stage.  I might suggest the latter since it allows us to split the work of read thread among multiple CPUs since reading shapefiles can sometimes be a bottleneck.